Sources/ClientRuntime/Networking/Http/HttpContent.swift

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import AwsCommonRuntimeKit
+
+/// A wrapper around a StreamReader that conforms to the IStreamable protocol
+/// This is used to pass the StreamReader to the CRT HttpMessage
+public class HttpContent: IStreamable {
+    internal let stream: StreamReader
+
+    public init(stream: StreamReader) {
+        self.stream = stream
+    }
+
+    public func seek(offset: Int64, streamSeekType: AwsCommonRuntimeKit.StreamSeekType) throws {
+        switch streamSeekType {
+        case .begin:
+            try stream.seek(offset: Int(offset))
+        case .end:
+            fatalError("This operation is not supported")
```

Comment:
Can we provide a more descriptive error message here? While we will hit this during debugging, stack traces aren't always accurate in production and it'd be good to easily identify this case from the error message.

Reply:
sure